document-generation/ifp-document.ts

```diff
@@ -0,0 +1,34 @@
+import { logger } from "../utils/logging";
+import createReport from "docx-templates";
+import { IncrementalFundingPlan } from "../models/document-generation";
+import { ApiBase64SuccessResponse, SuccessStatusCode } from "../utils/response";
+import { getFundingDocInfo } from "./utils/utils";
+
+export async function generateIFPDocument(
+  template: Buffer,
+  payload: IncrementalFundingPlan
+): Promise<ApiBase64SuccessResponse> {
+  const report = Buffer.from(
+    await createReport({
+      template,
+      data: {
+        ...payload,
+        // update amounts to have commas
+        estimatedTaskOrderValue: payload.estimatedTaskOrderValue.toLocaleString("en-US"),
+        initialAmount: payload.initialAmount.toLocaleString("en-US"),
+        remainingAmount: payload.remainingAmount.toLocaleString("en-US"),
+        // construct Funding Document information to be used in template
+        fundingDocInfo: getFundingDocInfo(payload.fundingDocument),
+      },
+      cmdDelimiter: ["{", "}"],
+    })
+  );
```

Comment:
Should there be a `try... catch` in case `createReport` results in an error?

Reply:
yeah, good point. I know the `errorHandlerMiddleware` and `middy` end up capturing most of the errors, so I sometimes leave out the `try/catch` blocks.

Reply:
True, middy will probably catch it and return a 500 back to SNOW. No need to add try/catch in that case.